src/cltl/combot/infra/di_container.py

```diff
@@ -0,0 +1,64 @@
+from threading import Lock
+from time import sleep
+
+_MAX_WAIT = 1000
+
+
+class DIContainer(object):
```

Comment:
If you are OK with that I can take a look and remove the usage of this entirely (outside of this PR..).